ground/gcs/src/plugins/uavtalk/telemetry.h

```diff
@@ -84,7 +84,9 @@ class Telemetry : public QObject
     Telemetry(UAVTalk *utalk, UAVObjectManager *objMngr);
     ~Telemetry();
     TelemetryStats getStats();
-    QByteArray *downloadFile(quint32 fileId, quint32 maxSize);
+    QByteArray *downloadFile(quint32 fileId, quint32 maxSize,
+            std::function<void(quint32)>progressCb = nullptr);
```

Comment:
I think QEventLoop is newer than a lot of the codebase, as are lambdas.  We also do relatively few things in the main UI  that are multistep-oriented... uploader, calibration, log download... and the wizards.  I think these would all be better written as eventloops (though cal should really be modal)